### DISCUSSION TITLE
LAN and BLE at the same time

### DISCUSSION THREAD

**Discussion by d-solis-tovar** [micropython/micropython#16150] [ESP32]
Hi,

I read some old posts about the infeasibility for LAN and BLE to be working at the same time in an ESP32, because they share hardware resources. As these posts were quite old, I would like to ask if with latest versions of micropython this is somehow possible now. 

Thank you in advance!

**Comment by bidrohini** [comment]
For concurrent network and Bluetooth applications,  you can consider ESP32 with dual-module setups. Some setups use one ESP32 for Wi-Fi/LAN and another for BLE.

**Reply by d-solis-tovar** [reply]
Thank you very much for your answer.

This could be a solution. Using any microcontroller with BLE capabilities (or even HM-10 BLE module) and adding some code for communication between microcontrollers (such as UART) will definitely work.

As this solution is more complex to implement than "single-module" solution, first I would like to be sure that currently it isn't possible for a single ESP32 to be running LAN and BLE at the same time. 

Can you confirm that?

**Reply by ma261065** [reply]
I have used an ESP32 to listen for BLE broadcasts and then send MQTT messages over WiFi, so it definitely is possible now to use both "concurrently".

Under the covers it is multiplexing the access to the single radio using the ESP-IDF coexistence feature - https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/coexist.html

**Reply by d-solis-tovar** [reply]
It's true that WiFi and BLE can work at the same time. The problem arises when using LAN (with LAN8720, W5500, etc. ethernet controller) and BLE at the same time. ESP32 crashes and reset.
